block/manager.go

```diff
@@ -836,6 +825,9 @@ func (m *Manager) publishBlock(ctx context.Context) error {
 			return err
 		}
 
+		block.SignedHeader.Validators = m.getLastStateValidators()
```

Comment:
Was the reason for moving these lines down so that in the event of an error generating the datahash, we aren't preemptively updating the validators? 

If so, please add that context in a comment so that the order of operations is understood and preserved.

Reply:
i don't think the order matters here.